heudiconv/parser.py

```diff
@@ -60,58 +62,87 @@ def find_files(regex, topdir=op.curdir, exclude=None,
             yield path
 
 
-def get_extracted_dicoms(fl):
-    """Given a list of files, possibly extract some from tarballs.
+def get_extracted_dicoms(
+        fl: typing.Iterable[str]
+        ) -> typing.ItemsView[typing.Optional[int], typing.List[str]]:
+    """Given a collection of files and/or directories, possibly extract 
+    some from tarballs.
 
-    For 'classical' heudiconv, if multiple tarballs are provided, they correspond
-    to different sessions, so here we would group into sessions and return
-    pairs  `sessionid`, `files`  with `sessionid` being None if no "sessions"
-    detected for that file or there was just a single tarball in the list
-    """
-    # TODO: bring check back?
-    # if any(not tarfile.is_tarfile(i) for i in fl):
-    #     raise ValueError("some but not all input files are tar files")
+    Parameters
+    ----------
+    fl : Iterable[str]
+        An iterable (e.g., list or tuple) of files to process.
+
+    Returns
+    -------
+    ItemsView[int | None, list[str]]
```

Comment:
Unfortunately on this aspect there even were not discussion so I initiated one https://github.com/numpy/numpydoc/issues/196#issuecomment-1488734277 . I guess we are doomed to keep it as this for now here.